mods/ra/maps/allies-10a/allies10a.lua

```diff
@@ -0,0 +1,220 @@
+	end)
+
+	Trigger.OnEnteredProximityTrigger(OreAmbushTrigger.CenterPosition, WDist.FromCells(5), function(actor, id)
+		if actor.Owner == Greece and actor.Type == "harv" and not Map.LobbyOption("difficulty") == "easy" then
+			Trigger.RemoveProximityTrigger(id)
+			local ambush = Reinforcements.Reinforce(USSR, AmbushTeam, { OreAmbushEntry.Location})
+			Utils.Do(ambush, IdleHunt)
+		end
+	end)
+
+	local northFootTriggered
+	Trigger.OnEnteredFootprint(NorthHarassFootprint, function(actor, id)
+		if actor.Owner == Greece and not northFootTriggered then
+			Trigger.RemoveFootprintTrigger(id)
+			northFootTriggered = true
+
+			local northHarass = Reinforcements.Reinforce(USSR, NorthHarassTeam, { OreAmbushEntry.Location})
+			Utils.Do(northHarass, IdleHunt)
+		end
+	end)
+end
+
+LaunchMissiles = function()
+	MissilesLaunched = true
+	local missileCam = Actor.Create("camera", true, { Owner = Greece, Location = FCom.Location })
+	Camera.Position = FCom.CenterPosition
+	Media.PlaySpeechNotification(Greece, "AbombLaunchDetected")
+	MissileSilo1.ActivateNukePower(CPos.New(127, 127))
+
+	Trigger.AfterDelay(DateTime.Seconds(3), function()
+		Media.DisplayMessage("INCOMING TRANSMISSION", "LANDCOM 16")
+		Media.PlaySpeechNotification(Greece, "AbombLaunchDetected")
+		MissileSilo2.ActivateNukePower(CPos.New(127, 127))
+	end)
+
+	Trigger.AfterDelay(DateTime.Seconds(5), function()
+		Media.PlaySpeechNotification(Greece, "AbombLaunchDetected")
+		MissileSilo3.ActivateNukePower(CPos.New(127, 127))
+	end)
+
+	Trigger.AfterDelay(DateTime.Seconds(6), function()
+		Media.PlaySpeechNotification(Greece, "AbombLaunchDetected")
+		MissileSilo4.ActivateNukePower(CPos.New(127, 127))
+	end)
+
+	Trigger.AfterDelay(DateTime.Seconds(8), function()
+		local fmvStart = DateTime.GameTime
+		Media.PlayMovieFullscreen("ally10b.vqa", function()
+			-- Completing immediately indicates that the FMV is not available
+			-- Fall back to a text message
+			if fmvStart == DateTime.GameTime then
+				Media.DisplayMessage("Commander, we're tracking four missiles. They must be deactivated! We are scrambling a team to assult the missile control bunker. Clear the way and capture the enemy command center. Hurry!", "LANDCOM 16")
+			end
+		end)
+	end)
+
+	Trigger.AfterDelay(DateTime.Seconds(9), function()
+		CaptureFCom = Greece.AddObjective("Capture the enemy Command Center. Hurry!")
+		StartTimer = true
+		Media.PlaySpeechNotification(Greece, "TimerStarted")
+		Greece.MarkCompletedObjective(ApproachBase)
+	end)
+
+	Trigger.OnCapture(CommandCenter, function()
+		Greece.MarkCompletedObjective(CaptureFCom)
+	end)
+
+	Paradrop()
+	Trigger.AfterDelay(DateTime.Minutes(2), SendParabombs)
+end
+
+SilosDamaged = function()
+	if not MissilesLaunched then
+		LaunchMissiles()
+	end
+end
+
+Tick = function()
+	USSR.Cash = 50000
+	BadGuy.Cash = 50000
+
+	if StartTimer then
+		if ticked > 0 then
+			UserInterface.SetMissionText("Missiles reach their targets in " .. Utils.FormatTime(ticked), TimerColor)
```

Comment:
Was this supposed to be `Utils.FormatTime(TimerTicks - ticked)`? Otherwise it doesn't make much sense and confused me, because the label said "Missiles reach their targets in 55 minutes", but the audio cues said "5 minutes remaining". I suggest you use the TimeLimitManager API instead, it was built just for these use cases.

But in any case, isn't giving the players 59 minutes and 42 seconds (huh?) just to capture the FCOM a bit much? Unless they blundered into the base with a scout early in the mission, I guess. Was that the idea?

Reply:
The 59:42 timer comes from the video that plays when the player triggers the nuke-launch. If we convert OG time to OpenRA, the timer would be ~36 minutes. I'm a bit torn on that as players (who have videos installed) will see an hour timer in the video and 36 minutes right after.

This is the first of a two-part mission, so in theory the remaining time is supposed to carry over to the next one, though I'm not sure how/if the OG handled that.

Reply:
Ah, I missed that. That makes sense then. Re: that carry-over ability, didn't @reaperrr write something for that some time ago?